src/cdk-experimental/clipboard/clipboard.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {Inject, Injectable} from '@angular/core';
+
+/**
+ * A service for copying text to the clipboard.
+ *
+ * Example usage:
+ *
+ * clipboard.copy("copy this text");
+ */
+@Injectable({providedIn: 'root'})
+export class Clipboard {
+  private _document: Document;
+
+  constructor(@Inject(DOCUMENT) document: any) {
+    this._document = document;
+  }
+
+  /**
+   * Copies the provided text into the user's clipboard.
+   *
+   * @param text The string to copy.
+   * @return Whether the operation was successful.
```

Comment:
nit: we use `@returns` on GitHub (which is admittedly arbitrarily different from inside Google)

Reply:
Good to know! Done.